src/main/java/br/com/vaasschool/controller/dto/AdminController.java

```diff
@@ -28,7 +31,8 @@ String showDashboard() {
 
     @GetMapping("/admin/dashboard")
     String dashboard(Model model) {
-        CourseProjection instructorWithGreaterNumberOfCourses = courseRepository.findInstructorWithMoreCourses();
+        CourseProjection instructorWithGreaterNumberOfCourses = courseRepository.findInstructorWithMoreCourses()
+                .orElseThrow(() -> new ResponseStatusException(NOT_FOUND));
```

Comment:
Eu não lançaria uma exception, já que não é obrigatório existir um instrutor com maior número de cursos (muito menos lançaria um not found). Eu acataria esse optional e trataria isso na view...